backends/ebpf/bpfinclude/ebpf_map.h

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2018 VMware, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+ * This file defines a library of simple hashmap operations which emulate the behavior
+ * of the kernel ebpf map API. This library is currently not thread-safe.
+ */
+
+#ifndef BACKENDS_EBPF_BPFINCLUDE_EBPF_MAP_H_
+#define BACKENDS_EBPF_BPFINCLUDE_EBPF_MAP_H_
+
+#include "../contrib/uthash.h"  // exports string.h, stddef.h, and stdlib.h
+
+// BPF_ANY create new element or update existing
+// BPF_NOEXIST: create new element only if it didn't exist
+// BPF_EXIST: only update existing element
+enum bpf_flags { BPF_ANY, BPF_NOEXIST, BPF_EXIST};
```

Comment:
You can still put comments next to each value; some IDEs will pick them up and help when typing.